testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -83,25 +143,42 @@ async fn delete_k8s_collection<T: Clone + DeserializeOwned + Meta>(
     {
         either::Left(list) => {
             let names: Vec<_> = list.iter().map(Meta::name).collect();
-            println!("Deleting collection of {}: {:?}", name, names);
+            info!("Deleting collection of {}: {:?}", name, names);
         }
         either::Right(status) => {
-            println!("Deleted collection of {}: status={:?}", name, status);
+            info!("Deleted collection of {}: status={:?}", name, status);
         }
     }
 
     Ok(())
 }
 
-pub(crate) async fn delete_k8s_cluster() -> Result<()> {
+pub(crate) async fn delete_k8s_cluster(kube_namespace: String) -> Result<()> {
     let client: K8sClient = create_k8s_client().await;
-    let deployments: Api<Deployment> = Api::namespaced(client.clone(), "default");
-    let stateful_sets: Api<StatefulSet> = Api::namespaced(client, "default");
 
-    // delete all deployments and statefulsets
-    // cross this with all the compute resources created by aptos-node helm chart
-    delete_k8s_collection(deployments, "deployments").await?;
-    delete_k8s_collection(stateful_sets, "stateful_sets").await?;
+    // if operating on the default namespace,
+    match kube_namespace.as_str() {
+        "default" => {
+            let deployments: Api<Deployment> = Api::namespaced(client.clone(), "default");
+            let stateful_sets: Api<StatefulSet> = Api::namespaced(client, "default");
+
+            // delete all deployments and statefulsets
+            // cross this with all the compute resources created by aptos-node helm chart
+            delete_k8s_collection(deployments, "deployments").await?;
+            delete_k8s_collection(stateful_sets, "stateful_sets").await?;
```

Comment:
when you delete statefulsets you have to manually delete the PVCs / PVs that were created by the StatefulSets. It doesn't cascade delete them.
Another option to simplify this whole code is to simply disallow `default` as namespace for forge deployments altogether (don't think that's a good idea anyways). Simply catch on the CLI level and prevent using the `default` namespace.

Reply:
The `default` namespace is used by Forge to inherit helm values. I need to figure out a better way to share helm values, but until that's done, I'll keep the `default` namespace around. 

As for resource deletion, the idea is to delete all compute resources so that the cluster-autoscaler can kick in and spin down the cluster. It's true that the PVCs/PVs will be kept around, but they'll be re-created when forge is re-run in that namespace, or when we TF apply at a new era, so correctness is not at stake here, just about state cleanliness and infra cost (which I think at this point is not too much of an issue).

The other alternative is to just do a `helm uninstall --keep-history` like Forge was doing before, which will uninstall the compute and storage resources for us...

Reply:
@rustielin re infra cost, I think it sorta is. Afaik leftover EBS volumes is one of our largest bill items right now (see my slack thread).

Reply:
also would love to understand a bit better how it's reusing helm values from the default namespace. Don't quite understand this part yet.

Reply:
Ok updated to delete the unused PVCs. `helm uninstall` gets rid of the validator PVCs, but not fullnode ones (because fullnode ones are defined in the spec inline rather than as its own resource). The rest are removed manually.

```
$ cargo run -p forge-cli -- operator clean-up --namespace default
...
2022-07-05T23:58:49.015880Z [main] INFO testsuite/forge/src/backend/k8s/cluster_helper.rs:203 ["uninstall", "--namespace", "default", "--keep-history", "aptos-node"]
2022-07-05T23:58:49.948077Z [main] INFO testsuite/forge/src/backend/k8s/cluster_helper.rs:203 ["uninstall", "--namespace", "default", "--keep-history", "genesis"]
2022-07-05T23:58:51.370872Z [main] INFO testsuite/forge/src/backend/k8s/cluster_helper.rs:146 Deleting collection of pvcs: ["fn-aptos-node-0-fullnode-e10-0", "fn-aptos-node-1-fullnode-e10-0", "fn-aptos-node-10-fullnode-e10-0", "fn-aptos-node-11-fullnode-e10-0", "fn-aptos-node-12-fullnode-e10-0", "fn-aptos-node-13-fullnode-e10-0", "fn-aptos-node-14-fullnode-e10-0", "fn-aptos-node-15-fullnode-e10-0", "fn-aptos-node-16-fullnode-e10-0", "fn-aptos-node-17-fullnode-e10-0", "fn-aptos-node-18-fullnode-e10-0", "fn-aptos-node-19-fullnode-e10-0", "fn-aptos-node-2-fullnode-e10-0", "fn-aptos-node-20-fullnode-e10-0", "fn-aptos-node-21-fullnode-e10-0", "fn-aptos-node-22-fullnode-e10-0", "fn-aptos-node-23-fullnode-e10-0", "fn-aptos-node-24-fullnode-e10-0", "fn-aptos-node-25-fullnode-e10-0", "fn-aptos-node-26-fullnode-e10-0", "fn-aptos-node-27-fullnode-e10-0", "fn-aptos-node-28-fullnode-e10-0", "fn-aptos-node-29-fullnode-e10-0", "fn-aptos-node-3-fullnode-e10-0", "fn-aptos-node-4-fullnode-e10-0", "fn-aptos-node-5-fullnode-e10-0", "fn-aptos-node-6-fullnode-e10-0", "fn-aptos-node-7-fullnode-e10-0", "fn-aptos-node-8-fullnode-e10-0", "fn-aptos-node-9-fullnode-e10-0"]
2022-07-05T23:58:51.371247Z [main] INFO testsuite/forge/src/backend/k8s/cluster_helper.rs:310 aptos-node resources for Forge removed in namespace: default
```